### PR TITLE
Fixed transparency check and made new tests

### PR DESCRIPTION
**Pull request by simsine**
This is to fix the issue described in #51 aswell as to introduce new tests for uploading transparent avatars

### CODE REVIEW

File: phoenixRest/views/user/instance/__init__.py [hunk 1]
```diff
@@ -496,8 +496,8 @@ def upload_avatar(context, request):
     # Crop the image
     im = im.crop((x, y, x+w, y+h))
 
-    # Deal with RGBA
-    if im.mode in ('RGBA', 'LA') or (im.mode == 'P' and 'transparency' in im.info):
```
Comment (petterroea) [inline phoenixRest/views/user/instance/__init__.py:500]: won't things crash on save even if the mode is P but no transparency?

**Review by petterroea — CHANGES_REQUESTED** [review]

File: phoenixRest/views/user/instance/__init__.py (hunk 1 above)
Comment (petterroea) [reply phoenixRest/views/user/instance/__init__.py:500]: I thought the error was that PIL can't save a P image to JPEG(you have to convert it to RGB first)

Comment (simsine) [reply phoenixRest/views/user/instance/__init__.py:500]: The mode being P is what implies transparency here, converting and then saving an image sith P mode is perfectly fine. As for the transparency key in the info part, none of the images i have parsed have contained this flag and i am not sure where you have gotten this from?

Comment (simsine) [reply phoenixRest/views/user/instance/__init__.py:500]: > I thought the error was that PIL can't save a P image to JPEG(you have to convert it to RGB first)

Yes the code already does this, this is not the issue here

Comment (petterroea) [reply phoenixRest/views/user/instance/__init__.py:500]: https://pillow.readthedocs.io/en/stable/handbook/concepts.html

According to pillow docs, P is just "Palette" and PA is "Palette with alpha".

Comment (simsine) [reply phoenixRest/views/user/instance/__init__.py:500]: I see, i'll try making a PA mode PBG

Comment (petterroea) [reply phoenixRest/views/user/instance/__init__.py:500]: transparent_p.png is not transparent. You can double check this with `exiftool`.

**Review by petterroea — APPROVED** [review]
Good stuff, thanks!

**Pull request merged**